app/controllers/users_controller.rb

```diff
@@ -1,4 +1,11 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user,   only: [:edit, :update]
+  before_action :admin_user,     only: :destroy
+
+  def index
+    @users = User.paginate(page: params[:page])
```

Comment:
Read more about LIMIT and OFFSET, https://www.postgresql.org/docs/9.3/queries-limit.html